### PR TITLE
Add recipe for gz-sim

### PR DESCRIPTION
**Pull request by traversaro**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
macOS is failing with:
~~~
2022-10-15T15:15:48.3263600Z export PREFIX=/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh
2022-10-15T15:15:48.3348180Z export SRC_DIR=/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp
2022-10-15T15:15:50.0824780Z import: 'gz.sim7'
2022-10-15T15:15:53.0880990Z Fatal Python error: PyMUTEX_LOCK(_PyRuntime.ceval.gil.mutex) failed
2022-10-15T15:15:53.0982340Z 
2022-10-15T15:15:53.1085450Z /Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp/conda_test_runner.sh: line 3:   670 Abort trap: 6           "/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/python" -s "/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp/run_test.py"
2022-10-15T15:15:54.8870870Z Tests failed for gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2 - moving package to /Users/runner/mambaforge/conda-bld/broken
2022-10-15T15:15:54.8971060Z WARNING:conda_build.build:Tests failed for gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2 - moving package to /Users/runner/mambaforge/conda-bld/broken
2022-10-15T15:15:55.0518360Z TESTS FAILED: gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2
~~~

This is the same failure we experienced in https://github.com/conda-forge/libignition-math4-feedstock/pull/25#issuecomment-1114233306 and is discussed upstream in https://github.com/pybind/pybind11/issues/3081 . However, we never encountered it in the rest of gz-packages with Python bindings (i.e. gz-math7 and sdformat13).

**Comment by traversaro** [comment]
> macOS is failing with:
> 
> ```
> 2022-10-15T15:15:48.3263600Z export PREFIX=/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh
> 2022-10-15T15:15:48.3348180Z export SRC_DIR=/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp
> 2022-10-15T15:15:50.0824780Z import: 'gz.sim7'
> 2022-10-15T15:15:53.0880990Z Fatal Python error: PyMUTEX_LOCK(_PyRuntime.ceval.gil.mutex) failed
> 2022-10-15T15:15:53.0982340Z 
> 2022-10-15T15:15:53.1085450Z /Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp/conda_test_runner.sh: line 3:   670 Abort trap: 6           "/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/python" -s "/Users/runner/mambaforge/conda-bld/gz-sim7_1665843708985/test_tmp/run_test.py"
> 2022-10-15T15:15:54.8870870Z Tests failed for gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2 - moving package to /Users/runner/mambaforge/conda-bld/broken
> 2022-10-15T15:15:54.8971060Z WARNING:conda_build.build:Tests failed for gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2 - moving package to /Users/runner/mambaforge/conda-bld/broken
> 2022-10-15T15:15:55.0518360Z TESTS FAILED: gz-sim7-python-7.0.0-py37h4e80f62_0.tar.bz2
> ```
> 
> This is the same failure we experienced in [conda-forge/libignition-math4-feedstock#25 (comment)](https://github.com/conda-forge/libignition-math4-feedstock/pull/25#issuecomment-1114233306) and is discussed upstream in [pybind/pybind11#3081](https://github.com/pybind/pybind11/issues/3081) . However, we never encountered it in the rest of gz-packages with Python bindings (i.e. gz-math7 and sdformat13).

Probably the correct fix (missing in gz-sim) is:
* https://github.com/gazebosim/gz-math/pull/497
* https://github.com/gazebosim/sdformat/pull/1127

**Comment by traversaro** [comment]
Approval of being mantainers from the mantainers (i.e. GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.)
* @wolfv:  https://github.com/conda-forge/staged-recipes/pull/20547#issuecomment-1261908808
* @Tobias-Fischer: https://github.com/conda-forge/staged-recipes/pull/20593#issuecomment-1262678266

@conda-forge/help-c-cpp The recipe is ready for review.

**Comment by traversaro** [comment]
Side note: on macOS there are several warnings:
~~~
2022-10-16T20:56:41.0614120Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0615880Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0617010Z   path: 'include/X11/DECkeysym.h'
2022-10-16T20:56:41.0617340Z 
2022-10-16T20:56:41.0617640Z 
2022-10-16T20:56:41.0618150Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0619310Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0620370Z   path: 'include/X11/HPkeysym.h'
2022-10-16T20:56:41.0620670Z 
2022-10-16T20:56:41.0620850Z 
2022-10-16T20:56:41.0621440Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0622670Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0623760Z   path: 'include/X11/Sunkeysym.h'
2022-10-16T20:56:41.0624100Z 
2022-10-16T20:56:41.0624350Z 
2022-10-16T20:56:41.0624890Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0626100Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0627290Z   path: 'include/X11/X.h'
2022-10-16T20:56:41.0627700Z 
2022-10-16T20:56:41.0627980Z 
2022-10-16T20:56:41.0628560Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0629790Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0630930Z   path: 'include/X11/XF86keysym.h'
2022-10-16T20:56:41.0631280Z 
2022-10-16T20:56:41.0631480Z 
2022-10-16T20:56:41.0631980Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0633800Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0634980Z   path: 'include/X11/Xatom.h'
2022-10-16T20:56:41.0635330Z 
2022-10-16T20:56:41.0635590Z 
2022-10-16T20:56:41.0636140Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0637390Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0638510Z   path: 'include/X11/Xfuncproto.h'
2022-10-16T20:56:41.0638880Z 
2022-10-16T20:56:41.0639120Z 
2022-10-16T20:56:41.0639680Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0640910Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0642030Z   path: 'include/X11/ap_keysym.h'
2022-10-16T20:56:41.0642380Z 
2022-10-16T20:56:41.0642610Z 
2022-10-16T20:56:41.0643180Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0644410Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0645950Z   path: 'include/X11/keysym.h'
2022-10-16T20:56:41.0646290Z 
2022-10-16T20:56:41.0646550Z 
2022-10-16T20:56:41.0647090Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0648360Z   packages: conda-forge/osx-64::xorg-xproto-7.0.31-h35c211d_1007, conda-forge/osx-64::tk-8.6.12-h5dbffcc_0
2022-10-16T20:56:41.0649470Z   path: 'include/X11/keysymdef.h'
2022-10-16T20:56:41.0649830Z 
2022-10-16T20:56:41.0650070Z 
2022-10-16T20:56:41.0650620Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0651840Z   packages: conda-forge/osx-64::tk-8.6.12-h5dbffcc_0, conda-forge/osx-64::xorg-libx11-1.7.2-h0d85af4_0
2022-10-16T20:56:41.0652950Z   path: 'include/X11/Xlib.h'
2022-10-16T20:56:41.0653280Z 
2022-10-16T20:56:41.0653520Z 
2022-10-16T20:56:41.0654180Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0655400Z   packages: conda-forge/osx-64::tk-8.6.12-h5dbffcc_0, conda-forge/osx-64::xorg-libx11-1.7.2-h0d85af4_0
2022-10-16T20:56:41.0656580Z   path: 'include/X11/Xutil.h'
2022-10-16T20:56:41.0656910Z 
2022-10-16T20:56:41.0657170Z 
2022-10-16T20:56:41.0657730Z ClobberWarning: This transaction has incompatible packages due to a shared path.
2022-10-16T20:56:41.0658960Z   packages: conda-forge/osx-64::tk-8.6.12-h5dbffcc_0, conda-forge/osx-64::xorg-libx11-1.7.2-h0d85af4_0
2022-10-16T20:56:41.0660060Z   path: 'include/X11/cursorfont.h'
2022-10-16T20:56:41.0660430Z 
2022-10-16T20:56:41.0660660Z 
2022-10-16T20:56:41.0660890Z 
2022-10-16T20:56:41.0661300Z done
~~~

however, this is already tracked in https://github.com/conda-forge/tk-feedstock/issues/15 .

File: recipes/gz-sim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        - {{ pin_subpackage(python_name, max_pin='x.x.x') }}
+    test:
+      imports:
+        - gz.{{ component_name }}{{ major_version }}
+
+about:
+  home: https://github.com/gazebosim/gz-sim
+  license: Apache-2.0
+  license_file: LICENSE
+  summary: "Open source robotics simulator. The latest version of Gazebo."
```
Comment (j-rivero) [inline recipes/gz-sim/meta.yaml:140]:
```suggestion
  summary: "Open source robotics simulator. The 7 version of Gazebo Sim."
```
minor thing expecting new version in the future.

File: recipes/gz-sim/build_cxx.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+if [[ "${CONDA_BUILD_CROSS_COMPILATION}" == "1" ]]; then
+  export CMAKE_ARGS="${CMAKE_ARGS} -DProtobuf_PROTOC_EXECUTABLE=$BUILD_PREFIX/bin/protoc"
+fi
+
+mkdir build
+cd build
+
+cmake ${CMAKE_ARGS} -GNinja .. \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=True \
```
Comment (j-rivero) [inline recipes/gz-sim/build_cxx.sh:17]: We probably don't need this in linux:
```
2022-10-16T20:14:15.0348427Z CMake Warning:
2022-10-16T20:14:15.0351812Z   Manually-specified variables were not used by the project:
2022-10-16T20:14:15.0354384Z 
2022-10-16T20:14:15.0356888Z     CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP
```

Comment (traversaro) [reply recipes/gz-sim/build_cxx.sh]: Thanks, done in https://github.com/conda-forge/staged-recipes/pull/20686/commits/3cafedcec82fe37c2b9d8e74e35fbcc87df13c41 .

File: recipes/gz-sim/meta.yaml (hunk 1 above)
Comment (j-rivero) [reply recipes/gz-sim/meta.yaml:140]: > Probably we can remove that sentence altogether?

+1

**Review by j-rivero — APPROVED** [review]

**Comment by traversaro** [comment]
Thanks @wolfv!

**Pull request merged**